sdks/go.mod

```diff
@@ -20,7 +20,7 @@
 // directory.
 module github.com/apache/beam/sdks/v2
 
-go 1.20
+go 1.21
```

Comment:
this is required for builtin max.

go 1.20 has already eol so we should bump it anyways.

Reply:
A single use of `max` isn't sufficient cause to bump the minimum SDK version IMO. I'd recommend just writing the standard if instead.

Reply:
While I *don't* think it'll be a problem to do it as a rule, there's just other work that might be good to do when bumping the minimum version as well, and that's very out of scope of this PR.

Basically, bumping the minimum version is a *huge* change just to avoid a single if statement, compared to writing the conditional.